networkx/convert_matrix.py

```diff
@@ -1046,6 +1047,11 @@ def from_numpy_array(A, parallel_edges=False, create_using=None, edge_attr="weig
         The attribute to which the array values are assigned on each edge. If
         it is None, edge attributes will not be assigned.
 
+    node_labels : list of nodes, optional
+        A list of objects to use as the nodes in the graph. If provided, the
```

Comment:
maybe `sequence` instead of `list`? There might be a better word too. In this PR we use a pandas `Series` and the object `range(n)` so if there is a word to describe those as well as a list or a tuple (or I suppose even a dict)

Reply:
Agreed - the important element is an order-preserving iterable, which I think sequence fits the bill best. There may be finer distinctions but it's at least 80% of the way there!